### PR TITLE
Add websocketTimeout to RoomConnectOptions

### PR DESCRIPTION
**Pull request by lukasIO**

### CODE REVIEW

File: src/api/SignalClient.ts [hunk 1]
```diff
@@ -224,9 +217,16 @@ export class SignalClient {
     return new Promise<JoinResponse | ReconnectResponse | void>(async (resolve, reject) => {
       const abortHandler = async () => {
         this.close();
+        clearTimeout(wsTimeout);
         reject(new ConnectionError('room connection has been cancelled (signal)'));
       };
 
+      console.log('timeout', opts.websocketTimeout);
```
Comment (davidzhao) [inline src/api/SignalClient.ts:224]: console to remove

File: src/connectionHelper/checks/websocket.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ export class WebSocketCheck extends Checker {
       autoSubscribe: true,
       maxRetries: 0,
       e2eeEnabled: false,
+      websocketTimeout: 15_000,
```
Comment (davidzhao) [inline src/connectionHelper/checks/websocket.ts:20]: should it just use default?

**Review by davidzhao — APPROVED** [review]
lgtm!

File: src/connectionHelper/checks/websocket.ts (hunk 2 above)
Comment (lukasIO) [reply src/connectionHelper/checks/websocket.ts:20]: the reason for it being explicit here is that the default opts get applied on the room level, not on the client level. So the client API requires it to be present explicitly

File: src/api/SignalClient.ts (hunk 1 above)
Comment (lukasIO) [reply src/api/SignalClient.ts:224]: thanks!

**Pull request merged**